x/qgb/abci.go

```diff
@@ -113,3 +121,89 @@ func handleValsetRequest(ctx sdk.Context, k keeper.Keeper) {
 		}
 	}
 }
+
+// PruneIfNeeded runs basic checks on saved attestations to see if we need to prune or not.
+// Also, proposes some gas optimizations by adding more checks to return before running any
+// of the pruning logic.
+func PruneIfNeeded(ctx sdk.Context, k keeper.Keeper) {
```

Comment:
[question] 

1. Does this function need to be exported?
1. Are the gas optimizations an implementation detail? They don't appear to be "returned" from this function.

```go
// maybePruneAttestations prunes attestations from state if...
func maybePruneAttestations(ctx sdk.Context, k keeper.Keeper) {
```

Reply:
changed